app/controllers/katello/api/v2/content_uploads_controller.rb

```diff
@@ -11,12 +11,16 @@ class Api::V2::ContentUploadsController < Api::V2::ApiController
     param :size, :number, :required => true, :desc => N_("Size of file to upload")
     param :checksum, String, :required => false, :desc => N_("Checksum of file to upload")
     param :content_type, RepositoryTypeManager.uploadable_content_types(false).map(&:label), :required => false, :desc => N_("content type ('deb', 'docker_manifest', 'file', 'ostree', 'rpm', 'srpm')")
+    param :generic_content_type, String, :required => false, :desc => N_("Content type for generic content unit")
     def create
       content_type = params[:content_type] || ::Katello::RepositoryTypeManager.find(@repository.content_type).default_managed_content_type.label
-      RepositoryTypeManager.check_content_matches_repo_type!(@repository, content_type)
-
+      RepositoryTypeManager.check_content_matches_repo_type!(@repository, content_type) unless content_type == "generic"
```

Comment:
After some looking, I think we could fix `check_content_matches_repo_type!` to make it work for generic content types.